include/nbl/video/IDescriptorPool.h

```diff
@@ -29,12 +38,87 @@ class NBL_API IDescriptorPool : public core::IReferenceCounted, public IBackendO
             uint32_t count;
         };
 
-        explicit IDescriptorPool(core::smart_refctd_ptr<const ILogicalDevice>&& dev, uint32_t _maxSets) : IBackendObject(std::move(dev)), m_maxSets(_maxSets) {}
+        inline void createDescriptorSets(uint32_t count, const IGPUDescriptorSetLayout* const* layouts, core::smart_refctd_ptr<IGPUDescriptorSet>* output)
+        {
+            for (uint32_t i = 0u; i < count; ++i)
+            {
+                auto layout = core::smart_refctd_ptr<const IGPUDescriptorSetLayout>(layouts[i]);
+                output[i++] = createDescriptorSet(std::move(layout));
+            }
+        }
+        
+        core::smart_refctd_ptr<IGPUDescriptorSet> createDescriptorSet(core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& layout);
+        bool freeDescriptorSets(const uint32_t descriptorSetCount, IGPUDescriptorSet* const* const descriptorSets);
 
-        uint32_t getCapacity() const { return m_maxSets; }
+        inline uint32_t getCapacity() const { return m_maxSets; }
 
     protected:
+        explicit IDescriptorPool(core::smart_refctd_ptr<const ILogicalDevice>&& dev, const IDescriptorPool::E_CREATE_FLAGS flags, uint32_t _maxSets, const uint32_t poolSizeCount, const IDescriptorPool::SDescriptorPoolSize* poolSizes) : IBackendObject(std::move(dev)), m_maxSets(_maxSets), m_flags(flags)
```

Comment:
the API would be a lot cleaner if intstead of taking a pointer to an array of `SDescriptorPoolSize` we just made a struct
```cpp
struct CreateInfo
{
    core::bitflag<E_CREATE_FLAGS> flags = ECF_NONE;
    uint32_t maxSets = 0; 
    uint32_t maxDescriptorCount[asset::EDT_COUNT] = {0};
};
```

Reply:
and keep that around as a const member.

Reply:
Yes, indeed.

Resolved.